### PR TITLE
timex: unambiguous time conversion macros

### LINKED ISSUE #4211: timex: macro names are misleading
Am I the only one that thinks the name of `SEC_IN_USEC`, `SEC_IN_MS`, `MS_IN_USEC`, and `USEC_IN_NS` are somewhat missleading?
1. They are inconsistent in itself. Some abbreviate "seconds" with `SEC`, some with `S`
2. Their values reflect the opposite of their name: there are 1000000 microseconds in a second, not 1000000 seconds in a microsecond; there are 1000 milliseconds in a second, not 1000 seconds in a microsecond; etc.

### PR DESCRIPTION
**Pull request by OlegHahm**
Fixes #4211.

I used:
```shell
git grep -l SEC_IN_USEC | xargs sed -i 's/MIN_IN_SEC/SEC_PER_MIN/'
git grep -l SEC_IN_USEC | xargs sed -i 's/SEC_IN_CS/CS_PER_SEC/'
git grep -l SEC_IN_USEC | xargs sed -i 's/SEC_IN_MS/MS_PER_SEC/'
git grep -l SEC_IN_USEC | xargs sed -i 's/SEC_IN_MS/MS_PER_SEC/'
git grep -l SEC_IN_USEC | xargs sed -i 's/MS_IN_USEC/USEC_PER_MS/'
git grep -l SEC_IN_USEC | xargs sed -i 's/USEC_IN_NS/NS_PER_USEC/'
```

### CODE REVIEW

**Comment by kYc0o** [comment]
You forgot to replace `SEC_IN_USEC` ?

**Comment by kYc0o** [comment]
Well actually there are several errors... Can you try to fix them so I can rebase #6934 on top of this one? Or you prefer to first merge #6934 and then you add it to this PR?

**Comment by aabadie** [comment]
@kYc0o, I don't think both PR will conflict. If you use the right convention in #6394 it will be fine.

**Comment by OlegHahm** [comment]
I was a bit sloppy - as you can see from the commands pasted above. Should be fixed now by using the proper arguments for grep and adding a 'g' flag for `sed`.

**Comment by OlegHahm** [comment]
ccn-lite needs an update

**Comment by miri64** [comment]
> I used:
> ```shell
> git grep -l SEC_IN_USEC | xargs sed -i 's/MIN_IN_SEC/SEC_PER_MIN/'
> git grep -l SEC_IN_USEC | xargs sed -i 's/SEC_IN_CS/CS_PER_SEC/'
> git grep -l SEC_IN_USEC | xargs sed -i 's/SEC_IN_MS/MS_PER_SEC/'
> git grep -l SEC_IN_USEC | xargs sed -i 's/SEC_IN_MS/MS_PER_SEC/'
> git grep -l SEC_IN_USEC | xargs sed -i 's/MS_IN_USEC/USEC_PER_MS/'
> git grep -l SEC_IN_USEC | xargs sed -i 's/USEC_IN_NS/NS_PER_USEC/'
```

Wouldn't
```shell
git grep -l MIN_IN_SEC | xargs sed -i 's/MIN_IN_SEC/SEC_PER_MIN/'
git grep -l SEC_IN_CS | xargs sed -i 's/SEC_IN_CS/CS_PER_SEC/'
git grep -l SEC_IN_MS | xargs sed -i 's/SEC_IN_MS/MS_PER_SEC/'
git grep -l MS_IN_USEC | xargs sed -i 's/MS_IN_USEC/USEC_PER_MS/'
git grep -l USEC_IN_NS | xargs sed -i 's/USEC_IN_NS/NS_PER_USEC/'
```
make much more sense?

**Comment by OlegHahm** [comment]
see https://github.com/RIOT-OS/RIOT/pull/6399#issuecomment-273354580

**Comment by miri64** [comment]
Sorry :-/

**Comment by miri64** [comment]
But as the discussion does not provide a consensus in #4211: what about `CS` -> `CSEC`, `MS` -> `MSEC`, `NS` -> `NSEC`?

**Comment by OlegHahm** [comment]
I have no strong opinion, can do, if you prefer.

**Comment by miri64** [comment]
I prefer it, but you [[1]][1] and @BytesGalore [[2]][2] (quoting SI units) disagreed in #4211, with no reply to my argument why I still prefer it [[3]][3] (e.g. `US_PER_S` looks weird).

[1]: https://github.com/RIOT-OS/RIOT/issues/4211#issuecomment-153219374
[2]: https://github.com/RIOT-OS/RIOT/issues/4211#issuecomment-153071788
[3]: https://github.com/RIOT-OS/RIOT/issues/4211#issuecomment-153276004

**Comment by OlegHahm** [comment]
Actually, I interpreted #4211 as a consensus.

**Comment by miri64** [comment]
On the `_PER_` part, yes. But the discussion thread seems a little bit more ambiguous about the `S`/`SEC` discussion to me (later posts liking the `_PER_` only mention `USEC_PER_SEC`, the units of which are already `SEC`).

**Comment by OlegHahm** [comment]
Which I read as an agreement on this solution. The only one who voted for SI units was @BytesGalore who kept silent during the later discussion with can be interpreted as agreement.

How about we make a quick poll tonight?

**Comment by OlegHahm** [comment]
For the record, I would prefer SEC for seconds and MS, NS, US and CS for the rest - even if it is not 100% consistent.

**Comment by miri64** [comment]
> How about we make a quick poll tonight?

:+1:

**Comment by kYc0o** [comment]
>For the record, I would prefer SEC for seconds and MS, NS, US and CS for the rest - even if it is not 100% consistent.

👍

**Comment by miri64** [comment]
> For the record, I would prefer SEC for seconds and MS, NS, US and CS for the rest - even if it is not 100% consistent.

We talked about it offline, and I agree now.

**Comment by OlegHahm** [comment]
Updated accordingly.

**Comment by OlegHahm** [comment]
Squashed

**Comment by OlegHahm** [comment]
Oops, fixed it.

**Comment by smlng** [comment]
@OlegHahm build fails due to patch that is not required anymore, so remove `pkg/ccn-lite/patches/`, that should do and I think will also make my PR #6364 obsolete 😒

**Comment by OlegHahm** [comment]
Updated CCN-lite and removed patches. Ready to squash?

**Comment by cgundogan** [comment]
Jenkins says in https://riot-ci.inet.haw-hamburg.de/job/RIOT/job/PR-6399/9/artifact/error_airfy-beacon_linux_other_tests.log:
```
Building tests/xtimer_usleep for airfy-beacon
make: Entering directory '/opt/jenkins/workspace/RIOT_PR-6399-SI7A7I2QFQLIKR4ZMJNGNMT2GKJLP3TAHIUJWDXUDWBJZWO2Z3RA@2/tests/xtimer_usleep'
Building application "xtimer_usleep" for "airfy-beacon" with MCU "nrf51".

main.c: In function 'main':
main.c:37:28: error: 'SEC_IN_USEC' undeclared (first use in this function)
         xtimer_usleep(ONE_SEC_SLEEP);
                            ^
main.c:37:28: note: each undeclared identifier is reported only once for each function it appears in
```

**Comment by OlegHahm** [comment]
Fixed the new test. Should I squash?

**Comment by cgundogan** [comment]
👍 squash please

**Comment by OlegHahm** [comment]
squashed

**Comment by smlng** [comment]
needs rebase, now

**Comment by OlegHahm** [comment]
rebased

**Comment by smlng** [comment]
requires merge of [ccn-lite PR 104](https://github.com/cn-uofbasel/ccn-lite/issues/104) and update ccn-lite pkg_version here, afterwards

**Comment by miri64** [comment]
https://github.com/cn-uofbasel/ccn-lite/pull/104 was merged

File: pkg/ccn-lite/Makefile [hunk 1]
```diff
@@ -1,6 +1,6 @@
 PKG_NAME=ccn-lite
 PKG_URL=https://github.com/cn-uofbasel/ccn-lite/
-PKG_VERSION=0b1de2da1ef407ee5793c0e7eda420391ae056dc
+PKG_VERSION=3bddb34dbb27d814acaca922655d760481b2998a
```
Comment (miri64) [inline pkg/ccn-lite/Makefile]: Please update to https://github.com/cn-uofbasel/ccn-lite/commit/10119ca8173457e8a26b28fc9f30fe0d97d17857

**Comment by OlegHahm** [comment]
I don't understand what https://github.com/cn-uofbasel/ccn-lite/pull/104 has to do with this PR?

**Comment by smlng** [comment]
@OlegHahm nothing directly, but if you update the package version (which have to do) and make you PR get through CI. You need the fixed ccn-lite version, that is with pr 104 merged.

**Comment by smlng** [comment]
see error log from jenkins [here](https://riot-ci.inet.haw-hamburg.de/view/Pipelines/job/RIOT/job/PR-6399/11/artifact/error_samr21-xpro_linux_examples.log)

**Comment by OlegHahm** [comment]
Ah, I wasn't aware that https://github.com/cn-uofbasel/ccn-lite/commit/d720a6c4cd124994b1e8b48acbbcf78ff02fa741 was already merged.

**Comment by smlng** [comment]
@OlegHahm: please update pkg version, so we get this and macOS support (#6341) into the release?

**Comment by OlegHahm** [comment]
Updated

**Pull request merged**